packages/scripts/src/send-email-blast.js

```diff
@@ -0,0 +1,154 @@
+import fs from 'fs';
+import dotenv from 'dotenv';
+import { readFileSync } from 'fs';
+import aws from 'aws-sdk';
+import { PromisePool } from '@supercharge/promise-pool';
+
+import { subjectOne, templateOne } from './constants.js';
+dotenv.config({ path: '../../.env' });
+
+const MAX_RETRY = 5;
+let retryCount = 0;
+let errors = [];
+let sent = 0;
+
+// authenticate aws ses
+const credentials = new aws.Credentials({
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+  sessionToken: process.env.AWS_SESSION_TOKEN,
+});
+aws.config.credentials = credentials;
+aws.config.region = process.env.AWS_S3_REGION;
+
+const ses = new aws.SES();
+
+// genereates csv file for any email NOT sent
+function generateEmailCSV(array, outputFilePath) {
+  let fulltext = 'name,email\n';
+  array.map(row => {
+    fulltext += `${row.name},${row.email}\n`;
+  });
+  fs.writeFileSync(outputFilePath, fulltext);
+}
+
+// get emails from csv file
+async function getEmails() {
+  const file = readFileSync('./in/emails.csv').toString();
+  // Split the string into rows, discard the columns row
+  return file
+    .split('\n')
+    .slice(1)
+    .map((rowString, index) => {
+      // Split each row into it's variables
+      let row = rowString.split(',');
+      return {
+        name: row[0].trim(),
+        email: row[1].trim(),
+        confirmation_id: row[2].trim(),
+      };
+    });
+}
+
+// remove duplicate email entries
+function findDuplicates(emails) {
+  let unique = [];
+  let result = [];
+  emails.forEach(email => {
+    if (!unique.includes(email.email)) {
+      unique.push(email.email);
+      result.push(email);
+    } else {
+      console.log('Duplicate found', JSON.stringify(email));
+    }
+  });
+  return result;
+}
+
+// send individual emails using AWS SES
+async function sendEmail(user) {
+  const { name, email, confirmation_id } = user;
+  // unsure of actual template body, using hardcoded html string for now
+  const mailOptions = {
+    from: process.env.MAIL_FROM,
+    to: [email],
+    subject: subjectOne,
+    body: templateOne,
+  };
+
+  if (!ses) return;
+
+  const params = {
+    Destination: {
+      ToAddresses: [...mailOptions.to],
+    },
+    Message: {
+      Body: {
+        Html: {
+          Charset: 'UTF-8',
+          Data: mailOptions.body,
+        },
+      },
+      Subject: {
+        Charset: 'UTF-8',
+        Data: mailOptions.subject,
+      },
+    },
+    //TO-DO: testing purposes, need to remove and replace with prod
+    Source: 'EHPRDoNotReply@dev.ehpr.freshworks.club',
+  };
+
+  try {
+    console.log(`Trying to Send Email to ${email}`);
+    const mail = await ses.sendEmail(params).promise();
+    sent++;
+    return mail;
+  } catch (e) {
+    // set max retry count for failed emails
+    if (MAX_RETRY > retryCount) {
+      console.log(`Will try sending to ${user.email} ${MAX_RETRY - retryCount} more time(s)`);
```

Comment:
Does this message mean the retryCount should be local to this function?
Or retryCount would be an object of email as key.

Reply:
each email will have it's own retry count, so user1@email.com will have 5 tries, user2@email.com will have 5 tries etc.

Reply:
Do you mean each email has its own retry count with the current logic?

Reply:
not with the current logic no, I will fix that flaw you pointed out below!  But that is the intent